src/stdio.rs

```diff
@@ -101,40 +103,44 @@ impl_from_for_stdio!(process::ChildStdin);
 impl_from_for_stdio!(process::ChildStdout);
 impl_from_for_stdio!(process::ChildStderr);
 
+impl_from_for_stdio!(ChildStdin);
+impl_from_for_stdio!(ChildStdout);
+impl_from_for_stdio!(ChildStderr);
+
 impl_from_for_stdio!(File);
 
 macro_rules! impl_try_from_tokio_process_child_for_stdio {
-    ($type:ident, $wrapper:ty) => {
+    ($type:ident) => {
         impl TryFrom<tokio::process::$type> for Stdio {
             type Error = Error;
 
             fn try_from(arg: tokio::process::$type) -> Result<Self, Self::Error> {
-                let wrapper: $wrapper = arg.try_into()?;
+                let wrapper: $type = arg.try_into()?;
                 Ok(wrapper.0.into())
             }
         }
     };
 }
 
-impl_try_from_tokio_process_child_for_stdio!(ChildStdin, ChildInputWrapper);
-impl_try_from_tokio_process_child_for_stdio!(ChildStdout, ChildOutputWrapper);
-impl_try_from_tokio_process_child_for_stdio!(ChildStderr, ChildOutputWrapper);
+impl_try_from_tokio_process_child_for_stdio!(ChildStdin);
+impl_try_from_tokio_process_child_for_stdio!(ChildStdout);
+impl_try_from_tokio_process_child_for_stdio!(ChildStderr);
 
 /// Input for the remote child.
-pub type ChildStdin = tokio_pipe::PipeWrite;
+#[derive(Debug)]
+pub struct ChildStdin(tokio_pipe::PipeWrite);
 
 /// Stdout for the remote child.
-pub type ChildStdout = tokio_pipe::PipeRead;
+#[derive(Debug)]
+pub struct ChildStdout(tokio_pipe::PipeRead);
 
 /// Stderr for the remote child.
-pub type ChildStderr = tokio_pipe::PipeRead;
-
-pub(crate) struct ChildInputWrapper(pub(crate) ChildStdin);
-pub(crate) struct ChildOutputWrapper(pub(crate) ChildStderr);
+#[derive(Debug)]
+pub struct ChildStderr(tokio_pipe::PipeRead);
 
 macro_rules! impl_from_impl_child_io {
-    (process, $type:ident, $wrapper:ty) => {
-        impl TryFrom<tokio::process::$type> for $wrapper {
+    (process, $type:ident, $inner:ty) => {
+        impl TryFrom<tokio::process::$type> for $type {
```

Comment:
This assumes that our wrapper will always be called the same as `tokio`'s — is that an assumption we expect will continue to hold? I guess probably?

Reply:
I don't think we have any reason to change the name.

If we do, then it will be trivial to update this macro.